### PR TITLE
Add poetry for managing environment

### PR DESCRIPTION
**Pull request by davehunt**
Also reformatted using black.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[tool.poetry]
+name = "perf-triage"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
+license = "MPL-2.0"
+
+[tool.poetry.dependencies]
+python = "^3.9"
+google-auth-oauthlib = "^0.4.6"
```
Comment (mcomella) [inline pyproject.toml:10]: Should we remove requirements.txt?

File: README.md [hunk 2]
```diff
@@ -1,21 +1,24 @@
...
-The first time you run, you'll need to create a virtualenv and install the
-dependencies:
-```sh
-# Create virtualenv
-python3 -m venv venv
+For convenience, you can use poetry to manage dependencies and virtual environments.
 
-# Activate virtualenv
-source venv/bin/activate
+When running for the first time, you will need to [install poetry](https://python-poetry.org/docs/#installation) and then run `poetry install` to create the virtual environment and install dependencies.
```
Comment (mcomella) [inline README.md:8]: What are do we get out of poetry that we don't get out of venv? I haven't used it before. If we don't have a strong reason to use it over venv, I'd prefer to use venv to reduce our external dependencies, reducing our attack surface area (the target being our dev machines or CI systems).

**Review by mcomella — APPROVED** [review]
r+ – I have questions but no need to block.

File: pyproject.toml (hunk 1 above)
Comment (davehunt) [reply pyproject.toml:10]: Oh right, good catch!

File: README.md (hunk 2 above)
Comment (davehunt) [reply README.md:8]: Mostly just convenience. If you'd prefer to continue using venv directly that works for me. We can close this pull request.

Comment (mcomella) [reply README.md:8]: Up to you! I don't know poetry and who maintains it so I don't have any context to trust it.

Comment (davehunt) [reply README.md:8]: I'll close this for now, we can always revisit if it feels valuable in the future.

**Pull request closed without merging**